PR1

AlexanderSparrow opened the pull request:
Iteration1

AlexanderSparrow committed:
Финальное задание 10го спринта.
Реализация пунктов технического задания. Код написан. Тесты postman проходят на ура. Готов предьявить на суд ревьюера:)

kesch9 commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,29 @@
+    @Past(message = "Дата рождения не может быть в будущем")
+    private LocalDate birthday;
+
+    public String getName() {
```
Обязательно
Нужно именно заменить name на login, если имени нет, при создании и обновления пользователя. Добавляя метод getName у вас поле будет все равно пустым и вы вернете из контроллера пустое поле.

AlexanderSparrow replied:
Изменил логику UserController, чтобы при пустом имени оно менялось на логин

kesch9 commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,44 @@
+    private int nextId = 1;
+
+    @PostMapping
+    public User addUser(@Valid @RequestBody User user) {
```
Обязательно
По заданию еще нужно добавить unit тесты
_Добавьте unit-тесты для валидации моделей. Убедитесь, что она работает на граничных условиях._

AlexanderSparrow replied:
Unit  тесты классов модели добавлены

kesch9 reviewed (commented):
Нужно исправить ошибки сборки

AlexanderSparrow committed:
Финальное задание 10го спринта.
Реализация пунктов технического задания. Код написан. Тесты postman проходят на ура. Доработа логика UserController по обработке пользователя с незаполненным полем имя. Дописаны unit тесты, согласно ТЗ

AlexanderSparrow committed:
Финальное задание 10го спринта.
Исправлено Raw use of parameterized class 'Set'

AlexanderSparrow committed:
Финальное задание 10го спринта.
Исправлено ошибки CheckStyle : добавден ЖИРНЫЙ пробел

kesch9 reviewed (approved):
Approve
